library/std/src/sys/sync/condvar/no_threads.rs

```diff
@@ -1,25 +1,48 @@
+use crate::cell::RefCell;
 use crate::sys::sync::Mutex;
+use crate::thread::sleep;
 use crate::time::Duration;
 
-pub struct Condvar {}
+pub struct Condvar {
+    value: RefCell<u32>,
+}
+
+unsafe impl Send for Condvar {}
+unsafe impl Sync for Condvar {} // no threads on this platform
 
 impl Condvar {
     #[inline]
     pub const fn new() -> Condvar {
-        Condvar {}
+        Condvar { value: RefCell::new(0) }
     }
 
     #[inline]
-    pub fn notify_one(&self) {}
+    pub fn notify_one(&self) {
+        *self.value.borrow_mut() += 1;
+    }
 
     #[inline]
-    pub fn notify_all(&self) {}
+    pub fn notify_all(&self) {
+        *self.value.borrow_mut() += 1;
+    }
```

Comment:
Maybe the type should be `Saturating<u32>` so this has no chance of wrapping to 0.